File: Previous Contest/3502. Minimum Cost to Reach Every Position/Solution.java
import java.lang.Math;

class Solution {
    public int[] minCosts(int[] costs) {
        // N : len(cost)
        // Time Complexity: O(N)
        // Space Complexity: O(N) -> with the output arr
        int n = costs.length;
        int[] minCostArr = new int[n];
        for (int i = 0; i < n; i++) {
            minCostArr[i] = (i != 0) ? Math.min(minCostArr[i - 1], costs[i]) : costs[i];
        }
        return minCostArr;
    }
}

File: Previous Contest/3502. Minimum Cost to Reach Every Position/Solution.py
from typing import List

class Solution:
    def minCosts(self, costs: List[int]) -> List[int]:
        # N : len(cost)
        # Time Complexity: O(N)
        # Space Complexity: O(N) -> with the output arr
        minCostList = []
        for cost in costs:
            cost = min(minCostList[-1], cost) if minCostList else cost
            minCostList.append(cost)
        return minCostList